src/Lexer.php

```diff
@@ -318,12 +322,8 @@ private function lexExpression(): void
             $this->moveCursor($match[0]);
         }
         // numbers
-        elseif (preg_match(self::REGEX_NUMBER, $this->code, $match, 0, $this->cursor)) {
-            $number = (float) $match[0];  // floats
-            if (ctype_digit($match[0]) && $number <= \PHP_INT_MAX) {
-                $number = (int) $match[0]; // integers lower than the maximum
-            }
-            $this->pushToken(/* Token::NUMBER_TYPE */ 6, $number);
+        elseif (preg_match(self::REGEX_NUMBER, $this->code, $match, 0, $this->cursor) && ('.' !== $match[0][0] || !$this->lastToken || !\in_array($this->lastToken->getType(), [/* Token::NAME_TYPE */ 5, /* Token::NUMBER_TYPE */ 6], true))) {
```

Comment:
Don't we have the same issue in ExpressionLanguage?

Reply:
IIRC I checked it and the issue doesn't exist in ExpressionLanguage because `.` can only be used for accessing public properties and for calling methods but both cannot start with a number.